### This is the course website for Ecological and Evolutionary Genomics 2018

#### The course is taught by Eric Collins (@rec3141), Assistant Professor of Biological Oceanography at the University of Alaska Fairbanks

#### We are using GitHub as a repository for code developed in class and for background material related to the class.

#### Quick link to the EEG2018 Team Discussion board: 
- <https://github.com/orgs/Ecological-and-Evolutionary-Genomics/teams/eeg2018>

### Bash shell and command line environment
- using BioLinux  
    - Lecture 1: <https://github.com/Ecological-and-Evolutionary-Genomics/eeg2018/wiki/Day-1---Jan-29---BioLinux-introduction>
- bash scripting  
    - Exercise 1: <https://github.com/Ecological-and-Evolutionary-Genomics/eeg2018/wiki/Exercise-1---Jan-29---bash-shell-introduction>
    - Exercise 2: <https://github.com/Ecological-and-Evolutionary-Genomics/eeg2018/wiki/Exercise-2---Feb-8---bash-scripting>

### Working in the cloud
- using Github for version control
- using PATRIC for genome and metagenome analysis
    - Exercise 3: <https://github.com/Ecological-and-Evolutionary-Genomics/eeg2018/wiki/Exercise-3---Feb-12---PATRIC>
- using IMG and MG-RAST for comparative genomics and metagenomics
- using Galaxy for comparative genomics and metagenomics 
- using the Arctic Region Supercomputing Center for taxonomic identification of short reads

### Taxonomic identification  
- from amplicon sequencing of ribosomal RNA genes, with mothur
    - Exercise 4: <https://github.com/Ecological-and-Evolutionary-Genomics/eeg2018/wiki/Exercise-4---Feb-19---mothur>
- from metagenomes, with k-mers using Kaiju
- from metagenomes, with conserved core genes using Phylosift

### Similarity searching
- using BLAST and DIAMOND

### Genome Assembly and Annotation
- genome assembly with SPAdes
- genome annotation with RAST

### Metagenome Assembly and Binning
- metagenome assembly with metaSPADES
- metagenome binning with metaBAT
- metagenome bin visualizaiton with VizBin, anvi'o?
